src/AbstractOperations/binary_operations.jl

```diff
@@ -29,12 +29,15 @@ end
 # Recompute location of binary operation
 @inline at(loc, β::BinaryOperation) = β.op(loc, at(loc, β.a), at(loc, β.b))
 
+indices(β::BinaryOperation) = interpolate_indices(β.a, β.b)
```

Comment:
Doesn't `interpolate_indices` need to know about `▶a` and `▶b`? Because the indices of the operator depend on both the indices of the underlying field, and how that field is interpolated?

Reply:
(if we interpolate in the direction of the "slice")

Reply:
Alternatively it needs to know the location of the fields (that might be easier since you can design some logic around multiple dispatch, potentially saving some code)